src/Mappers/InventoryMapper.php

```diff
@@ -34,85 +41,131 @@ public function getAvailableDate($mainVariationId) {
   }
 
   /**
-   * @param $stockList
+   * @param mixed $variationStock
    *
-   * @return array
+   * @return int|mixed
    */
-  private function getStockFromWarehouse($stockList) {
+  public function getNetStock($variationStock)
+  {
     /**
-     * @var WarehouseSupplierRepository $warehouseSupplierRepository
+     * @var AbstractConfigHelper $configHelper
      */
-    $warehouseSupplierRepository = pluginApp(WarehouseSupplierRepository::class);
-    $supplierId = null;
-    $warehouseId = null;
+    $configHelper = pluginApp(AbstractConfigHelper::class);
+    $stockBuffer = $configHelper->getStockBufferValue();
+    if ($variationStock->netStock > $stockBuffer) {
+      return $variationStock->netStock - $stockBuffer;
+    } else {
+      return 0;
+    }
+  }
+
+  /**
+   * Create an Inventory DTO for each Warehouse in the Variation data
+   * @param $variationData
+   *
+   * @return RequestDTO[]
+   */
+  public function createInventoryDTOsFromVariation($variationData)
+  {
+    /** @var LoggerContract $loggerContract */
+    $loggerContract = pluginApp(LoggerContract::class);
+
+    /** @var RequestDTO[] $inventoryDTOs */
+    $inventoryDTOs = [];
+
+    $supplierPartNumber = $this->getSupplierPartNumberFromVariation($variationData);
+
+    $mainVariationId = $variationData['id'];
+    $nextAvailableDate = $this->getAvailableDate($mainVariationId); // Pending. Need Item
+
+    $stockList = $variationData['stock'];
     foreach ($stockList as $stock) {
       $warehouseId = $stock['warehouseId'];
-      if (isset($warehouseId)) {
-        $warehouseSupplierMapping = $warehouseSupplierRepository->findByWarehouseId($warehouseId);
-        $supplierId = isset($warehouseSupplierMapping) ? $warehouseSupplierMapping->supplierId : null;
-        if (isset($supplierId)) {
-          $stock['supplierId'] = $supplierId;
-
-          return $stock;
-        }
+      if (!isset($warehouseId)) {
+        // we don't know the warehouse, so we can't figure out the supplier ID.
+        // Not an error, but unexpected.
+        $loggerContract->warning(
+          TranslationHelper::getLoggerKey(self::LOG_KEY_STOCK_MISSING_WAREHOUSE),
+          [
+            'additionalInfo' => ['variationID' => $mainVariationId,],
+            'method' => __METHOD__
+          ]
+        );
+        continue;
       }
+
+      // this is a 'mixed' value - might be null.
+      $supplierId = $this->getSupplierIDForWarehouseID($warehouseId);
+
+      if (!isset($supplierId) || $supplierId === 0 || $supplierId === '0') {
```

Comment:
not necessarily an issue - do we know if there are cases where getSupplierID returns 0/'0'

Reply:
no, this is just defensive coding.